components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/dao/AssociationDAO.java

```diff
@@ -107,9 +115,59 @@ public void updateAssociation(Association associationDTO)
 
     /**
      *
+     * @param tenantId
+     * @param filter
+     * @param offset
+     * @param limit
+     * @throws InternalWorkflowException
+     */
+    public List<Association> listPaginatedAssociations(int tenantId, String filter, int offset, int limit) throws InternalWorkflowException{
+
+        String sqlQuery;
+        List<Association> associations = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            String filterResolvedForSQL = resolveSQLFilter(filter);
+            sqlQuery = getSqlQuery();
+            try (PreparedStatement prepStmt = generatePrepStmt(connection, sqlQuery, tenantId,
+                    filterResolvedForSQL, offset, limit)) {
+                try (ResultSet resultSet = prepStmt.executeQuery()) {
+                    while (resultSet.next()) {
+                        String condition = resultSet.getString(SQLConstants.CONDITION_COLUMN);
+                        String eventId = resultSet.getString(SQLConstants.EVENT_ID_COLUMN);
+                        String associationId = String.valueOf(resultSet.getInt(SQLConstants.ID_COLUMN));
+                        String associationName = resultSet.getString(SQLConstants.ASSOCIATION_NAME_COLUMN);
+                        String workflowName = resultSet.getString(SQLConstants.WF_NAME_COLUMN);
+                        String isEnable = resultSet.getString(SQLConstants.ASSOCIATION_IS_ENABLED);
```

Comment:
can we use resultSet.getBoolean here and get as bool

Reply:
It is a char value in the db and returns '1' or '0'. not boolean